src/dune_lang/package_dependency.ml

```diff
@@ -40,12 +40,43 @@ let encode { name; constraint_ } =
   | Some c -> pair Package_name.encode Package_constraint.encode (name, c)
 ;;
 
+(* Check for common typos in package dependency constraints *)
+let check_for_version_typo { name; constraint_ } =
+  let open Package_constraint in
+  match constraint_ with
+  | Some (Uop (Relop.Eq, Value.String_literal "version")) ->
+    Some
+      ( sprintf
+          "Possible typo in package dependency for %s: '(= version)' might be a mistake."
```

Comment:
I think it would be more accurate to say "Possible typo in constraint for dependency %S: ...".